src/main.rs

```diff
@@ -313,7 +313,23 @@ fn construct_config(mut opts: Opts, pattern_regexps: &[String]) -> Result<Config
             .map(crate::fmt::FormatTemplate::parse),
         command: command.map(Arc::new),
         batch_size: opts.batch_size,
-        exclude_patterns: opts.exclude.iter().map(|p| String::from("!") + p).collect(),
+        exclude_patterns: opts
+            .exclude
+            .iter()
+            .map(|p| {
+                let path = Path::new(p);
```

Comment:
It isn't a path, it's a pattern.

What if there is a glob that occurs between the root and the search path?